packages/shared/src/hooks/usePostModalNavigation.ts

```diff
@@ -106,8 +114,18 @@ export const usePostModalNavigation = (
     };
   }, [items]);
 
+  const getPostPosition = () => {
+    const isPost = (item: FeedItem) => item.type === 'post';
+    const firstPost = items.findIndex(isPost);
+    const lastPost = [...items].reverse().findIndex(isPost);
+    const isLast = items.length - lastPost - 1 === openedPostIndex;
+    if (firstPost === openedPostIndex)
+      return isLast ? PostPosition.Only : PostPosition.First;
+    return isLast ? PostPosition.Last : PostPosition.Middle;
+  };
```

Comment:
Not sure if this will become heavy on people who scroll a lot, seems like it would calculate every new item, and with a long list it might be expensive?

Reply:
You are right. I think this is O(n²)
I will try to improve it